qa/integration-tests/src/test/java/io/camunda/it/utils/CamundaMultiDBExtension.java

```diff
@@ -0,0 +1,238 @@
+  private final TestStandaloneApplication<?> testApplication;
+  private String testPrefix;
+  private final MultiDbConfigurator multiDbConfigurator;
+
+  public CamundaMultiDBExtension() {
+    this(new TestSimpleCamundaApplication());
+    closeables.add(testApplication);
+    testApplication
+        .withBrokerConfig(cfg -> cfg.getGateway().setEnable(true))
+        .withExporter(
+            "recordingExporter", cfg -> cfg.setClassName(RecordingExporter.class.getName()));
+  }
+
+  public CamundaMultiDBExtension(final TestStandaloneApplication testApplication) {
+    this.testApplication = testApplication;
+    multiDbConfigurator = new MultiDbConfigurator(testApplication);
+    // resolve active database and exporter type
+    final String property = System.getProperty(PROP_CAMUNDA_IT_DATABASE_TYPE);
+    databaseType =
+        property == null ? DatabaseType.LOCAL : DatabaseType.valueOf(property.toUpperCase());
+  }
+
+  @Override
+  public void beforeAll(final ExtensionContext context) {
+    LOGGER.info("Starting up Camunda instance, with {}", databaseType);
+    final Class<?> testClass = context.getRequiredTestClass();
+    testPrefix = testClass.getSimpleName().toLowerCase();
+
+    switch (databaseType) {
+      case LOCAL -> {
+        final ElasticsearchContainer elasticsearchContainer = setupElasticsearch();
+        final String elasticSearchUrl = "http://" + elasticsearchContainer.getHttpHostAddress();
+        validateESConnection(elasticSearchUrl);
+        multiDbConfigurator.configureElasticsearchSupport(elasticSearchUrl, testPrefix);
+      }
+      case ES -> {
+        validateESConnection(DEFAULT_ES_URL);
+        multiDbConfigurator.configureElasticsearchSupport(DEFAULT_ES_URL, testPrefix);
+      }
+      case OS ->
+          multiDbConfigurator.configureOpenSearchSupport(
+              DEFAULT_OS_URL, testPrefix, DEFAULT_OS_ADMIN_USER, DEFAULT_OS_ADMIN_PW);
+      case RDBMS -> multiDbConfigurator.configureRDBMSSupport();
+      default -> throw new RuntimeException("Unknown exporter type");
+    }
+    testApplication.start();
+    testApplication.awaitCompleteTopology();
+
+    injectFields(testClass, null, ModifierSupport::isStatic);
+  }
+
+  private ElasticsearchContainer setupElasticsearch() {
+    final ElasticsearchContainer elasticsearchContainer =
+        TestSearchContainers.createDefeaultElasticsearchContainer();
+    elasticsearchContainer.start();
+    closeables.add(elasticsearchContainer);
+    return elasticsearchContainer;
+  }
+
+  private static void validateESConnection(final String url) {
+    final HttpRequest httpRequest =
+        HttpRequest.newBuilder().GET().uri(URI.create(String.format("%s/", url))).build();
+    try (final HttpClient httpClient = HttpClient.newHttpClient()) {
+      final HttpResponse<String> response = httpClient.send(httpRequest, BodyHandlers.ofString());
+      final int statusCode = response.statusCode();
+      assert statusCode / 100 == 2
```

Comment:
:question: Why / 100? I don't get it :smile:

Reply:
So that 201 is ok :-D